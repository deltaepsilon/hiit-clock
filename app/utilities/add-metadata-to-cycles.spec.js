import addMetadtaToCycles from './add-metadata-to-cycles';
import getTimerCycles from './get-timer-cycles';
const tabata = require('../data/tabata.json');

describe('addMetadataToCycles', () => {
  describe('Tabata 20/40', () => {
    let result;

    beforeEach(() => {
      const periods = tabata.find(({ name }) => name == 'Tabata 20/40');
      const cycles = getTimerCycles(periods);

      result = addMetadtaToCycles(cycles);
    });

    it('should have 20 cycles', () => {
      expect(result.length).toEqual(20);
    });

    it('should match snapshot', () => {
      expect(result).toMatchSnapshot();
    });
  });
});
