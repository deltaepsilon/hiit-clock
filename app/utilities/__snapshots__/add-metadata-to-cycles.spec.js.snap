// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMetadataToCycles Tabata 20/40 should match snapshot 1`] = `
Array [
  Array [
    Object {
      "cycleColor": "#f00081",
      "name": "Round 1",
      "percentOfCycle": 0.333,
      "periodColor": "#f00081",
      "periodIndex": 0,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#f00081",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 1,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ff5400",
      "name": "Round 2",
      "percentOfCycle": 0.333,
      "periodColor": "#ff5400",
      "periodIndex": 2,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ff5400",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 3,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ffbd00",
      "name": "Round 3",
      "percentOfCycle": 0.333,
      "periodColor": "#ffbd00",
      "periodIndex": 4,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ffbd00",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 5,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#109648",
      "name": "Round 4",
      "percentOfCycle": 0.333,
      "periodColor": "#109648",
      "periodIndex": 6,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#109648",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 7,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#1d4e89",
      "name": "Round 5",
      "percentOfCycle": 0.333,
      "periodColor": "#1d4e89",
      "periodIndex": 8,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#1d4e89",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 9,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#d15eff",
      "name": "Round 6",
      "percentOfCycle": 0.333,
      "periodColor": "#d15eff",
      "periodIndex": 10,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#d15eff",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 11,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#f00081",
      "name": "Round 7",
      "percentOfCycle": 0.333,
      "periodColor": "#f00081",
      "periodIndex": 12,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#f00081",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 13,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ff5400",
      "name": "Round 8",
      "percentOfCycle": 0.333,
      "periodColor": "#ff5400",
      "periodIndex": 14,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ff5400",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 15,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ffbd00",
      "name": "Round 9",
      "percentOfCycle": 0.333,
      "periodColor": "#ffbd00",
      "periodIndex": 16,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ffbd00",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 17,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#109648",
      "name": "Round 40",
      "percentOfCycle": 0.333,
      "periodColor": "#109648",
      "periodIndex": 18,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#109648",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 19,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#1d4e89",
      "name": "Round 11",
      "percentOfCycle": 0.333,
      "periodColor": "#1d4e89",
      "periodIndex": 20,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#1d4e89",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 21,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#d15eff",
      "name": "Round 12",
      "percentOfCycle": 0.333,
      "periodColor": "#d15eff",
      "periodIndex": 22,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#d15eff",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 23,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#f00081",
      "name": "Round 13",
      "percentOfCycle": 0.333,
      "periodColor": "#f00081",
      "periodIndex": 24,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#f00081",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 25,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ff5400",
      "name": "Round 14",
      "percentOfCycle": 0.333,
      "periodColor": "#ff5400",
      "periodIndex": 26,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ff5400",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 27,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ffbd00",
      "name": "Round 15",
      "percentOfCycle": 0.333,
      "periodColor": "#ffbd00",
      "periodIndex": 28,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ffbd00",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 29,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#109648",
      "name": "Round 16",
      "percentOfCycle": 0.333,
      "periodColor": "#109648",
      "periodIndex": 30,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#109648",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 31,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#1d4e89",
      "name": "Round 17",
      "percentOfCycle": 0.333,
      "periodColor": "#1d4e89",
      "periodIndex": 32,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#1d4e89",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 33,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#d15eff",
      "name": "Round 18",
      "percentOfCycle": 0.333,
      "periodColor": "#d15eff",
      "periodIndex": 34,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#d15eff",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 35,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#f00081",
      "name": "Round 19",
      "percentOfCycle": 0.333,
      "periodColor": "#f00081",
      "periodIndex": 36,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#f00081",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 37,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "cycleColor": "#ff5400",
      "name": "Round 20",
      "percentOfCycle": 0.333,
      "periodColor": "#ff5400",
      "periodIndex": 38,
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "cycleColor": "#ff5400",
      "name": "Rest",
      "percentOfCycle": 0.667,
      "periodColor": "#4f4f4f",
      "periodIndex": 39,
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
]
`;
